torchtext/data/datasets_utils.py

```diff
@@ -310,3 +311,97 @@ def __iter__(self):
                     columns[i].append(column)
         if len(columns) > 0:
             yield columns
+
+
+@functional_datapipe("parse_cnndm_data")
```

Comment:
For both of the `parse_cnndm` datapipes, let's mention in docstrings that the code is inspired from https://github.com/abisee/cnn-dailymail/blob/master/make_datafiles.py